File: protocol/src/test/java/quan/protocol/role/RoleInfo.java
package quan.protocol.role;

import java.util.HashSet;
import java.util.Arrays;
import java.io.IOException;
import quan.protocol.stream.WritableStream;
import java.util.HashMap;
import quan.protocol.Bean;
import quan.protocol.stream.ReadableStream;
import java.util.ArrayList;

/**
 * 角色信息
 * Created by {@link quan.protocol.generator.JavaGenerator}
 */
public class RoleInfo extends Bean {

    private long roleId;//角色id
    private boolean bo;
    private byte by;
    private short s;
    private int i;
    private float f;
    private double d;
    private String roleName;//角色名
    private RoleType roleType;
    private byte[] data;
    private ArrayList<Integer> list;
    private HashSet<Integer> set;
    private HashMap<Integer, Integer> map;

    public RoleInfo() {
        roleId = 111L;
        bo = true;
        by = (byte) 22;
        s = (short) 22;
        i = 11;
        f = 22.33332F;
        d = 33.332432D;
        roleName = "zhangsan";
        roleType = RoleType.type1;
        data = new byte[0];
        list = new ArrayList<>();
        set = new HashSet<>();
        map = new HashMap<>();
    }

    public long getRoleId() {
        return roleId;
    }

    public void setRoleId(long roleId) {
        this.roleId = roleId;
    }

    public boolean getBo() {
        return bo;
    }

    public void setBo(boolean bo) {
        this.bo = bo;
    }

    public byte getBy() {
        return by;
    }

    public void setBy(byte by) {
        this.by = by;
    }

    public short getS() {
        return s;
    }

    public void setS(short s) {
        this.s = s;
    }

    public int getI() {
        return i;
    }

    public void setI(int i) {
        this.i = i;
    }

    public float getF() {
        return f;
    }

    public void setF(float f) {
        this.f = f;
    }

    public double getD() {
        return d;
    }

    public void setD(double d) {
        this.d = d;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public RoleType getRoleType() {
        return roleType;
    }

    public void setRoleType(RoleType roleType) {
        this.roleType = roleType;
    }

    public byte[] getData() {
        return data;
    }

    public void setData(byte[] data) {
        this.data = data;
    }

    public ArrayList<Integer> getList() {
        return list;
    }

    public HashSet<Integer> getSet() {
        return set;
    }

    public HashMap<Integer, Integer> getMap() {
        return map;
    }


    @Override
    public void serialize(WritableStream writable) throws IOException {
        writable.writeLong(roleId);
        writable.writeBool(bo);
        writable.writeByte(by);
        writable.writeShort(s);
        writable.writeInt(i);
        writable.writeFloat(f);
        writable.writeDouble(d);
        writable.writeString(roleName);
        writable.writeInt(roleType.getValue());
        writable.writeBytes(data);
        writable.writeInt(list.size());
        for (int listValue : list) {
            writable.writeInt(listValue);
        }
        writable.writeInt(set.size());
        for (int setValue : set) {
            writable.writeInt(setValue);
        }
        writable.writeInt(map.size());
        for (int mapKey : map.keySet()) {
            writable.writeInt(mapKey);
            writable.writeInt(map.get(mapKey));
        }
    }

    @Override
    public void parse(ReadableStream readable) throws IOException {
        roleId = readable.readLong();
        bo = readable.readBool();
        by = readable.readByte();
        s = readable.readShort();
        i = readable.readInt();
        f = readable.readFloat();
        d = readable.readDouble();
        roleName = readable.readString();
        roleType = RoleType.valueOf(readable.readInt());
        data = readable.readBytes();
        int listSize = readable.readInt();
        for (int i = 0; i < listSize; i++) {
            list.add(readable.readInt());
        }
        int setSize = readable.readInt();
        for (int i = 0; i < setSize; i++) {
            set.add(readable.readInt());
        }
        int mapSize = readable.readInt();
        for (int i = 0; i < mapSize; i++) {
            map.put(readable.readInt(), readable.readInt());
        }
    }

    @Override
    public String toString() {
        return "RoleInfo{" +
                "roleId=" + roleId +
                ",bo=" + bo +
                ",by=" + by +
                ",s=" + s +
                ",i=" + i +
                ",f=" + f +
                ",d=" + d +
                ",roleName='" + roleName+ '\'' +
                ",roleType=" + roleType +
                ",data=" + Arrays.toString(data) +
                ",list=" + list +
                ",set=" + set +
                ",map=" + map +
                '}';

    }

}

File: message/message_cs/test/role/RoleInfo.cs
using System;
using System.Collections.Generic;
using message_cs;
using Buffer = message_cs.Buffer;

namespace message_cs.test.role
{
	/// <summary>
	/// 角色信息<br/>
	/// Created by 自动生成
	/// </summary>
    public class RoleInfo : Bean
    {
		public long roleId { get; set; }

		public bool bo { get; set; }

		public byte by { get; set; }

		public short s { get; set; }

		public int i { get; set; }

		public float f { get; set; }

		public double d { get; set; }

		private string _roleName = "";

		public string roleName
		{
	    	get => _roleName;
	    	set => _roleName = value ?? throw new NullReferenceException();
		}

		public RoleType roleType { get; set; }

		private byte[] _data = new byte[0];
        
		public byte[] data
		{
            get => _data;
            set => _data = value ?? throw new NullReferenceException();
		}

		public List<int> list { get; } = new List<int>();

		public HashSet<int> set { get; } = new HashSet<int>();

		public Dictionary<int, int> map { get; } = new Dictionary<int, int>();


		public RoleInfo()
		{
		}

		public override void Encode(Buffer buffer)
		{
	    	base.Encode(buffer);

		    buffer.WriteLong(roleId);
		    buffer.WriteBool(bo);
		    buffer.WriteByte(by);
		    buffer.WriteShort(s);
		    buffer.WriteInt(i);
		    buffer.WriteFloat(f);
		    buffer.WriteDouble(d);
		    buffer.WriteString(roleName);
			buffer.WriteInt((int)roleType);
		    buffer.WriteBytes(data);

		    buffer.WriteInt(list.Count);
		    foreach (var _list_Value in list) {
			    buffer.WriteInt(_list_Value);
		    }

		    buffer.WriteInt(set.Count);
		    foreach (var _set_Value in set) {
			    buffer.WriteInt(_set_Value);
		    }

		    buffer.WriteInt(map.Count);
		    foreach (var _map_Key in map.Keys) {
		        buffer.WriteInt(_map_Key);
			    buffer.WriteInt(map[_map_Key]);
		    }

		}

		public override void Decode(Buffer buffer)
		{
	    	base.Decode(buffer);

		    roleId = buffer.ReadLong();
		    bo = buffer.ReadBool();
		    by = buffer.ReadByte();
		    s = buffer.ReadShort();
		    i = buffer.ReadInt();
		    f = buffer.ReadFloat();
		    d = buffer.ReadDouble();
		    roleName = buffer.ReadString();
		    roleType = (RoleType)buffer.ReadInt();
		    data = buffer.ReadBytes();

		    var _list_Size = buffer.ReadInt();
		    for (var _index_ = 0; _index_ < _list_Size; _index_++) {
			    list.Add(buffer.ReadInt());
		    }

		    var _set_Size = buffer.ReadInt();
		    for (var _index_ = 0; _index_ < _set_Size; _index_++) {
			    set.Add(buffer.ReadInt());
		    }

		    var _map_Size = buffer.ReadInt();
		    for (var _index_ = 0; _index_ < _map_Size; _index_++) {
			    map.Add(buffer.ReadInt(), buffer.ReadInt());
		    }

		}

		public override string ToString()
		{
			return "RoleInfo{" +
					"roleId=" + roleId +
					",bo=" + bo +
					",by=" + by +
					",s=" + s +
					",i=" + i +
					",f=" + f +
					",d=" + d +
					",roleName='" + roleName + '\'' +
					",roleType=" + roleType +
					",data=" + Convert.ToBase64String(data) +
					",list=" + ToString(list) +
					",set=" + ToString(set) +
					",map=" + ToString(map) +
					'}';
		}
    }
}